### PR TITLE
Add tests for callback group

### PR DESCRIPTION
**Pull request by liquor233**
> [!IMPORTANT]  
> The `Update branch` button must only be pressed in very rare occassions.
> An outdated branch is never blocking the merge of a PR.
> Please reach out to the automation team before pressing that button.

# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog

- Add specific line by line info of high level changes in this PR.

# Usage

- You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

File: nemo/core/classes/modelPT.py [hunk 1]
```diff
@@ -48,7 +48,7 @@
 from nemo.core.classes.common import Model
 from nemo.core.connectors.save_restore_connector import SaveRestoreConnector
 from nemo.core.optim import McoreDistributedOptimizer, prepare_lr_scheduler
-from nemo.lightning.pytorch.callbacks.callback_group import CallbackGroup
```
Comment (PytLab) [inline nemo/core/classes/modelPT.py:51]: @liquor233 what is the reason to move callback group and base callback out of the pytorch/callbacks ?

File: tests/lightning/test_callbacks_group.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+    d = Dummy()
+    assert d.x == 1
+    assert start.called
+    assert end.called
+    # Flag indicating wrapping applied
+    assert getattr(Dummy.__init__, '_callback_group_wrapped', False) is True
+
+
+def test_hook_class_init_with_callbacks_idempotent():
```
Comment (PytLab) [inline tests/lightning/test_callbacks_group.py:170]: @liquor233 for an e2e / user facing test, we should test the class inheritance case

File: tests/lightning/test_one_logger_callback.py [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+from unittest.mock import MagicMock, patch
+
+import pytest
+from lightning.pytorch.callbacks import Callback as PTLCallback
+
+from nemo.lightning.base_callback import BaseCallback
+from nemo.lightning.one_logger_callback import OneLoggerNeMoCallback
+
+
+class TestOneLoggerNeMoCallback:
```
Comment (PytLab) [inline tests/lightning/test_one_logger_callback.py]: @liquor233 as we are have the base_callback module under lightning it is a generic base class not for one-logger. I think it is better to still keep one-logger callback related tests under `nv-one-logger` dir.
